### PR TITLE
gtk: specify template as an attribute of CompositeTemplate

### PR DESCRIPTION
**Pull request by pbor**
The source can be a file, a resource, or a string.

WIP: because I only quickly tested the file case using the example file

Also, it would be great to automagically hook into `class_init` instead of requiring to call `template_init`, but I have no idea on how to do that

### CODE REVIEW

File: gtk3-macros/src/composite_template_derive.rs [hunk 1]
```diff
@@ -45,6 +93,16 @@ pub fn impl_composite_template(input: &syn::DeriveInput) -> TokenStream {
     let name = &input.ident;
     let crate_ident = crate_ident_new();
 
+    let source = match parse_template_source(&input) {
+        Ok(v) => v,
+        Err(e) => abort_call_site!(
+            "{}: derive(CompositeTemplate) requires #[template(ui = \"/resource/path/name.ui\")]",
```
Comment (pbor) [inline gtk3-macros/src/composite_template_derive.rs:99]: error needs fixing

**Review by sdroege — APPROVED** [review]
Seems fine to me, and it has lots of useful docs now :)

**Comment by sdroege** [comment]
CI is unhappy though:

```
error[E0599]: no method named `init_template` found for reference `&InitializingObject<ExApplicationWindow>` in the current scope
   --> examples/src/bin/composite_template.rs:56:17
    |
56  |             obj.init_template();
    |                 ^^^^^^^^^^^^^ method not found in `&InitializingObject<ExApplicationWindow>`
    | 
   ::: /__w/gtk-rs/gtk-rs/glib/src/subclass/types.rs:383:1
    |
383 | pub struct InitializingObject<T: ObjectType>(Borrowed<T>);
    | ----------------------------------------------------------
    | |
    | doesn't satisfy `_: IsA<gtk::Widget>`
    | doesn't satisfy `_: gtk::WidgetExt`
    |
    = note: the method `init_template` exists but the following trait bounds were not satisfied:
            `InitializingObject<ExApplicationWindow>: IsA<gtk::Widget>`
            which is required by `InitializingObject<ExApplicationWindow>: gtk::WidgetExt`
            `&InitializingObject<ExApplicationWindow>: IsA<gtk::Widget>`
            which is required by `&InitializingObject<ExApplicationWindow>: gtk::WidgetExt`

error: aborting due to previous error
```

**Review by bilelmoussaoui — APPROVED** [review]

**Comment by ids1024** [comment]
> Also, it would be great to automagically hook into class_init instead of requiring to call template_init, but I have no idea on how to do that

I think this may be possible through some abuse of specialization... but that won't be stable any time soon.

**Comment by bilelmoussaoui** [comment]
@sdroege @GuillaumeGomez can we get this one merged so that both gtk3 & gtk4 composite templates support are synced

**Comment by sdroege** [comment]
Ah I thought I merged it already. Sorry!

**Pull request merged**